internal/consensus/reactor.go

```diff
@@ -1165,13 +1190,14 @@ func (ps *PeerState) SetHasProposalBlockPart(height int64, round int32, index in
 }
 
 func (ps *PeerState) setHasProposalBlockPart(height int64, round int32, index int) {
-	ps.logger.Debug("setHasProposalBlockPart",
-		"peerH/R",
-		log.NewLazySprintf("%d/%d", ps.PRS.Height, ps.PRS.Round),
-		"H/R",
-		log.NewLazySprintf("%d/%d", height, round),
-		"index", index)
-
+	if ps.logger.DebugOn() {
+		ps.logger.Debug("setHasProposalBlockPart",
```

Comment:
Are you sure that the lazy print does not solve the problem here?

Reply:
Yes, the lazy print is adding new objects to heap